cmd/kubelet/kubelet.go

```diff
@@ -32,29 +33,52 @@ import (
 	"k8s.io/kubernetes/cmd/kubelet/app"
 	"k8s.io/kubernetes/cmd/kubelet/app/options"
 	_ "k8s.io/kubernetes/pkg/client/metrics/prometheus" // for client metric registration
-	_ "k8s.io/kubernetes/pkg/version/prometheus"        // for version metric registration
+	"k8s.io/kubernetes/pkg/kubelet/apis/kubeletconfig"
+	_ "k8s.io/kubernetes/pkg/version/prometheus" // for version metric registration
 	"k8s.io/kubernetes/pkg/version/verflag"
 )
 
+func newFlagSet(kf *options.KubeletFlags, kc *kubeletconfig.KubeletConfiguration) *pflag.FlagSet {
+	fs := pflag.NewFlagSet(os.Args[0], pflag.ExitOnError)
+	// set the normalize func, similar to k8s.io/apiserver/pkg/util/flag/flags.go:InitFlags
+	fs.SetNormalizeFunc(flag.WordSepNormalizeFunc)
+	// explicitly add flags from libs that register global flags
+	options.AddGlobalFlags(fs)
+	// add the flags that target Kubelet objects
+	kf.AddFlags(fs)
+	options.AddKubeletConfigFlags(fs, kc)
+	return fs
+}
+
+func parseFlagSet(fs *pflag.FlagSet) error {
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		return err
+	}
+	fs.VisitAll(func(flag *pflag.Flag) {
+		glog.V(2).Infof("FLAG: --%s=%q", flag.Name, flag.Value)
```

Comment:
was leaving this intentional?

Reply:
Yeah, just to preserve the existing flag logging behavior. There may be better solutions.